functions/src/send-email-to-manager.ts

```diff
@@ -68,13 +68,18 @@ export const sendEmailToManager = functions.firestore
      */
     let emailSent = false;
     await firebase.firestore().runTransaction(async transaction => {
+      //Make sure we didn't send a mail already
```

Comment:
A la place de mettre un commentaire est ce qu'on pourrait plutôt faire un test pour éviter une potentielle régression ?

Reply:
We'd have to make the function pure first. I guess that would a good exercice.

Reply:
Updating a database can't be in a pure function right ?

I've done this for what it's worth
`await firebase.firestore().runTransaction(async transaction => {
      const transacVote = (await transaction.get(
        db.collection("vote").doc(voteSnapshot.id)
      )).data();
      if (transacVote && transacVote.emailToManagerSent == false) {
        await mailgunClient
          .messages()
          .send(message)
          .then(response => {
            transaction.update(voteSnapshot.ref, {
              emailToManagerSent: true,
              fullName: "*REDACTED*",
              email: "*REDACTED"
            });
          });
      }
    });`

Reply:
You're right that a pure function cannot update a database, but a pure function can return an instruction that the database must be updated.

However, in the case at hand, I'd say it is very hard to do because the correctness of the function relies on the transaction semantics of Firestore.